translate.js

```diff
@@ -0,0 +1,264 @@
+  text = text.replace(/\{:([^\}]+)\}/g, (match, p1, p2, offset, str) => {
+    pragmas.push(p1);
+    return `PRAGMAS${pragmas.length-1}`;
+  });
+
+  // Find special words and don't translate
+  text = text.replace(/\`([^\`]+)\`/g, (match, p1, p2, offset, str) => {
+    specialWords.push(p1);
+    return `__SPECIALWORD${specialWords.length-1}`;
+  });
+
+  const output = [];
+  let results = await translate.translate(text, {to});
+
+  let translations = results[0];
+  translations = Array.isArray(translations)
+    ? translations
+    : [translations];
+
+  // Note these fixes are not sustainable
+  translations.forEach((translation, i) => {
+
+    // Find markdown links that are broken [] () => []()
+    translation = translation.replace(/\[([^\]]+)\] \(([^\)]+)\)/g,'[$1]($2)');
+    // Find markdown links that are broken [] [] => [][]
+    translation = translation.replace(/\[([^\]]+)\] \[([^\]]+)\]/g,'[$1][$2]');
+    
+    // Clean up things that look like broken tags
+    translation = translation.replace(/<\/ ([^>]+)>/g,  (match, p1, p2, offset, str) => {
+      return `${match.replace(' ', '')}`;
+    });
+
+    // Find markdown image links that are broken ! []() => ![]()
+    translation = translation.replace(/! \[([^\]]+)\]\(([^\)]+)\)/g,' ![$1]($2)');
+    // Find markdown image links that are broken ! []() => ![][]
+    translation = translation.replace(/! \[([^\]]+)\]\[([^\]+)])/g,' ![$1][$2]');
+    // Find markdown links where the target has spaces in the wrong place [](/ ERROR /)
+    translation = translation.replace(/\[([^\]]+)\]\(\/( ([^\)]+) )\/\)/g,'[$1]($3)');
+    translation = translation.replace(/\[([^\]]+)\]\u{FF08}([^\u{FF09}]+)\u{FF09}/gu,'[$1]($2)');
+    translation = translation.replace(/＃/gu,'#');
+
+    // Remap all links of form []()
+    translation = translation.replace(/\[([^\]]+)\]\((\d+)\)/g, (match, p1, p2, offset, str) => {
+      return `[${p1}](${links.shift()})`;
+    });
+
+    // Remap all callouts
+    translation = translation.replace(/__SPECIALCALLOUTS(\d+)__/g, (match, p1, p2, offset, str) => {
+      return `${callouts.shift()} `;
+    });
+
+    // Remap all links of form [][]
+    translation = translation.replace(/\[([^\]]+)\]\[(\d+)\]/g, (match, p1, p2, offset, str) => {
+      return `[${p1}][${squareLinks.shift()}]`;
+    });
+
+    // Remap all {: } 
+    translation = translation.replace(/PRAGMAS(\d+)/, (match, p1, p2, offset, str) => {
+      return `{:${pragmas.shift()}}`;
+    });
+
+    // Remap all link defintions 
+    translation = translation.replace(/^__LINK_DEFS__ (\d+)/, (match, p1, p2, offset, str) => {
+      return `${linkDefs.shift()}`;
+    });
+
+    // Remap all src="" and src=''
+    translation = translation.replace(/SRCURL(\d+)/, (match, p1, p2, offset, str) => {
+      return `${srcs.shift()}`;
+    });
+
+    // Remap all specialWords 
+    translation = translation.replace(/__SPECIALWORD(\d+)/g, (match, p1, p2, offset, str) => {
```

Comment:
Not getting replaced properly - I see a bunch in the translated docs.

Reply:
good catch. will sort.